cpp/src/parquet/stream_writer.h

```diff
@@ -0,0 +1,185 @@
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PARQUET_STREAM_WRITER_H
+#define PARQUET_STREAM_WRITER_H
+
+#include <array>
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "arrow/util/string_view.h"
+#include "parquet/api/writer.h"
+
+namespace parquet {
+
+// A class for writing Parquet files using an output stream type API.
+//
+// The values given must be of the correct type i.e. the type must
+// match the file schema exactly otherwise a ParquetException will be
+// thrown.
+//
+// The user must explicitly indicate the end of the row using the
+// EndRow() function or EndRow output manipulator.
+//
+// A maximum row group size can be configured, the default size is
+// 512MB.  Alternatively the row group size can be set to zero and the
+// user can create new row groups by calling the EndRowGroup()
+// function or using the EndRowGroup output manipulator.
+//
+// Currently there is no support for optional or repeated fields.
+//
+class PARQUET_EXPORT StreamWriter {
+ public:
+  // N.B. Default constructed objects are not usable.  This
+  //      constructor is provided so that the object may be move
+  //      assigned afterwards.
+  StreamWriter() = default;
+
+  explicit StreamWriter(std::unique_ptr<ParquetFileWriter> File_Writer);
+
+  ~StreamWriter();
+
+  static void SetDefaultMaxRowGroupSize(const int64_t max_size);
+
+  void SetMaxRowGroupSize(const int64_t max_size);
+
+  // Moving is possible.
+  StreamWriter(StreamWriter&&) noexcept = default;
+  StreamWriter& operator=(StreamWriter&&) noexcept = default;
+
+  // Copying is not allowed.
+  StreamWriter(const StreamWriter&) = delete;
+  StreamWriter& operator=(const StreamWriter&) = delete;
+
+  StreamWriter& operator<<(const bool v);
+
+  StreamWriter& operator<<(const int8_t v);
+
+  StreamWriter& operator<<(const uint8_t v);
+
+  StreamWriter& operator<<(const int16_t v);
+
+  StreamWriter& operator<<(const uint16_t v);
+
+  StreamWriter& operator<<(const int32_t v);
+
+  StreamWriter& operator<<(const uint32_t v);
+
+  StreamWriter& operator<<(const int64_t v);
+
+  StreamWriter& operator<<(const uint64_t v);
+
+  StreamWriter& operator<<(const std::chrono::milliseconds& v);
+
+  StreamWriter& operator<<(const std::chrono::microseconds& v);
+
+  StreamWriter& operator<<(const float v);
+
+  StreamWriter& operator<<(const double v);
+
+  StreamWriter& operator<<(const char v);
+
+  // Helper class to write fixed length strings.
+  struct PARQUET_EXPORT FixedStringView {
```

Comment:
It's curious that this is required at all, could you comment why this helper class is needed when we have `arrow::util::string_view`?

Reply:
Yes and that class is used to write variable length string data.
However, another type is required to write fixed length data and so I think this helper class will be useful to distinguish this case.